README.md

```diff
@@ -1,407 +1,150 @@
+[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-data.csv.svg)](https://github.com/ballerina-platform/module-ballerina-data.csv/commits/master)
+[![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/data.csv.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fdata.csv)
 
-This library is the refined successor of the `ballerina/csvdata` module, incorporating enhanced functionalities and improved performance.
+The Ballerina CSV Data Library is a comprehensive toolkit designed to facilitate the handling and manipulation of CSV data within Ballerina applications. It streamlines the process of converting CSV data to native Ballerina data types, enabling developers to work with CSV content seamlessly and efficiently.
 
 ## Features
 
-- **Versatile CSV Data Input**: Accept CSV data as a csv, a string, byte array, or a stream and convert it into a Record value.
-- **CSV to Record Value Conversion**: Transform CSV data into Ballerina records with ease in compliance with OpenAPI 3 standards.
-- **Projection Support**: Perform selective conversion of CSV data subsets into Record values through projection.
+- **Versatile CSV Data Input**: Accept CSV data as a string, byte array, or a stream and convert it into a subtype of ballerina records or lists.
+- **CSV to anydata Value Conversion**: Transform CSV data into expected type which is subtype of ballerina records or lists.
+- **Projection Support**: Perform selective conversion of CSV data subsets into ballerina records or lists values through projection.
 
 ## Usage
 
-### Converting an CSV value to a Record value
+### Converting CSV string to a record array
 
-To convert an CSV value to a Record value, you can utilize the `fromCsvWithType` function provided by the library. The example below showcases the transformation of an CSV value into a Record value.
+To convert an CSV document value to a record value, you can utilize the `parseStringToRecord` function provided by the library. The example below showcases the transformation of a CSV document into a record array.
 
 ```ballerina
 import ballerina/data.csv;
 import ballerina/io;
 
-public function main() returns error? {
-    csv data = csv `<book>
-        <id>0</id>
-        <title>string</title>
-        <author>string</author>
-    </book>`;
+type Book record {
+    string name;
+    string author;
+    int year;
+};
 
-    Book book = check csvdata:fromCsvWithType(data);
-    io:println(book);
+public function main() returns error? {
+    string csvString = string `name,author,year
+                               Clean Code,Robert C. Martin,2008
+                               The Pragmatic Programmer,Andrew Hunt and David Thomas,1999`;
+
+    Book[] books = check csv:parseStringToRecord(csvString);
+    foreach var book in books {
+        io:println(book);
+    }
```

Comment:
```suggestion
    io:println(books);
```

Reply:
I think using foreach will be the better option in here.